src/timesync/timesyncd-manager.c

```diff
@@ -1131,3 +1138,60 @@ int manager_new(Manager **ret) {
 
         return 0;
 }
+
+static int manager_save_time_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        Manager *m = userdata;
+
+        assert(m);
+
+        return manager_save_time_and_rearm(m);
+}
+
+int manager_setup_save_time_event(Manager *m) {
+        int r;
+
+        assert(!m->event_save_time);
+
+        if (m->save_time_interval_usec == USEC_INFINITY)
+                return 0;
+
+        /* NB: we'll accumulate scheduling latencies here, but this doesn't matter */
+        r = sd_event_add_time_relative(
+                        m->event, &m->event_save_time,
+                        clock_boottime_or_monotonic(),
+                        m->save_time_interval_usec,
+                        10 * USEC_PER_SEC,
+                        manager_save_time_handler, m);
+        if (r < 0)
+                return log_error_errno(r, "Failed to add save time event: %m");
+
+        r = sd_event_source_set_enabled(m->event_save_time, SD_EVENT_ONESHOT);
+        if (r < 0)
+                return log_error_errno(r, "Failed to enable save time event: %m");
```

Comment:
This is not necessary, as `sd_event_add_time_relative()` enables the event source with `SD_EVENT_ONESHOT`.

Reply:
Not addressed, and you mistakenly dropped to reenable timer in manager_save_time_and_rearm()...

Reply:
@yuwata sorry, nuked the wrong part :exploding_head: